pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -85,15 +85,18 @@ func NewCmdStartBuild(fullName string, f *clientcmd.Factory, out io.Writer) *cob
 					}
 					// Iterate over watcher's results and search for
 					// the build we just started. Also make sure that
-					// the build is running
-					if build.Name == newBuild.Name && build.Status == buildapi.BuildStatusRunning {
-						rd, err := client.Get().Namespace(namespace).Prefix("redirect").Resource("buildLogs").Name(newBuild.Name).Stream()
-						checkErr(err)
-						defer rd.Close()
-
-						_, err = io.Copy(out, rd)
-						checkErr(err)
-						break
+					// the build is running or has failed
+					if build.Name == newBuild.Name {
+						switch build.Status {
+						case buildapi.BuildStatusRunning, buildapi.BuildStatusFailed:
```

Comment:
yes but couldn't it have transitioned through run and into complete before you do this check?  that's what's happening with failed...